src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedInteractiveInitializerMethod.cs

```diff
@@ -237,7 +237,15 @@ internal TypeSymbol ResultType
             {
                 diagnostics.Add(useSiteDiagnostic, NoLocation.Singleton);
             }
-            resultType = compilation.GetTypeByReflectionType(submissionReturnType, diagnostics);
+            // if no explicit return type is set on ScriptCompilationInfo, it will default to
+            // the _host_ corlib's System.Object (typeof(object)) and should never be null.
+            // In this case, redirect the host System.Object to the compilation's target
+            // corlib's System.Object. This allows cross compiling scripts to build on one
+            // corlib and run on another.
+            // cf. https://github.com/dotnet/roslyn/issues/8506
+            resultType = submissionReturnType == typeof(object)
```

Comment:
It feels to me that compilation.GetTypeByReflectionType should locate proper symbol without a need to special case System.Object. Otherwise, why other special types not special cased (System.Int32, for example)? Perhaps submissionReturnType is set to the wrong Type instance, perhaps that is what should be fixed?